lib/src/darwin.dart

```diff
@@ -431,6 +433,21 @@ class AVAudioSessionInterruptionOptions {
   int get hashCode => value.hashCode;
 }
 
+/// Temporary class to encapsulate the [AVAudioSessionRouteChangeReason] and
+/// a boolean to indicate if a previous route dropped had headphones port.
+/// Not according to documentation.
+class AVAudioSessionRouteChange {
+  final AVAudioSessionRouteChangeReason routeChangeReason;
+  final int _shouldInterrupt;
+
+  const AVAudioSessionRouteChange(
+      this.routeChangeReason, this._shouldInterrupt);
+
+  bool intToBool(int a) => a == 0 ? false : true;
+
+  bool get shouldInterrupt => intToBool(_shouldInterrupt);
+}
+
```

Comment:
This is the new class with the RouteChangeReason and a getter for if we want to interrupt.
Here would be preferable to have a list of outputs.